apps/TaskRunner/main.cpp

```diff
@@ -98,34 +145,49 @@ int main(int argc, char** argv) {
 
     ModuleEngine moduleEngine;
     moduleEngine.initialize();
+    LINFO("Initialization done.");
 
-    std::string tasksPath;
-    configuration.getValue(ConfigurationManager::KeyConfigTask, tasksPath);
+    // Parse commandline arguments
+    std::vector<std::string> args(argv, argv + argc);
 
-    LINFO("Initialization done.");
+    ghoul::cmdparser::CommandlineParser commandlineParser(
+        "OpenSpace TaskRunner",
+        ghoul::cmdparser::CommandlineParser::AllowUnknownCommands::Yes
+    );
 
-    TaskLoader taskLoader;
-    std::vector<std::unique_ptr<Task>> tasks = taskLoader.tasksFromFile(tasksPath);
-    
-    size_t nTasks = tasks.size();
-    if (nTasks == 1) {
-        LINFO("Task queue has 1 item");
-    } else {
-        LINFO("Task queue has " << tasks.size() << " items");
-    }
+    std::string tasksPath = "";
+    commandlineParser.addCommand(
+        std::make_unique<ghoul::cmdparser::SingleCommand<std::string>>(
+            &tasksPath,
+            "--task",
+            "-t",
+            "Provides the path to a task file to execute"
+            )
+    );
 
-    for (size_t i = 0; i < tasks.size(); i++) {
-        Task& task = *tasks[i].get();
-        LINFO("Performing task " << (i+1) << " out of " << tasks.size() << ": " << task.description());
-        ProgressBar progressBar(100);
-        auto onProgress = [&progressBar](float progress) {
-            progressBar.print(progress * 100);
-        };
-        task.perform(onProgress);
+    commandlineParser.setCommandLine(args);
+    commandlineParser.execute();
+
+    FileSys.setCurrentDirectory(launchDirectory);
+
+    if (tasksPath != "") {
+        performTasks(tasksPath);
+        return 0;
     }
 
-    std::cout << "Done performing tasks." << std::endl;
+    // If no task file was specified in as argument, run in CLI mode.
+
+    std::string tasksRoot;
+    if (configuration.getValue(ConfigurationManager::KeyConfigTasksRoot, tasksRoot)) {
+        LINFO("Task root: " << tasksRoot);
+        FileSys.setCurrentDirectory(ghoul::filesystem::Directory(absPath(tasksRoot)));
+    }
+    
+    while (true) {
+        std::cout << "TASK >";
+        std::cin >> tasksPath;
+        performTasks(tasksPath);
```

Comment:
Maybe checking `tasksPath` here for a `QUIT` so that we can stop the application?